components/gumroad/sources/common/common.mjs

```diff
@@ -0,0 +1,67 @@
+import gumroad from "../../gumroad.app.mjs";
+
+export default {
+  props: {
+    gumroad,
+    db: "$.service.db",
+    timer: {
+      type: "$.interface.timer",
+      static: {
+        intervalSeconds: 15 * 60, // 15 minutes
+      },
+    },
+  },
+  methods: {
+    emitEvent(data) {
+      throw new Error("emitEvent is not implemented", data);
+    },
+    getResources() {
+      throw new Error("getResources is not implemented");
+    },
+    getResourcesKey() {
+      throw new Error("getResourcesKey is not implemented");
+    },
+    _setLastTimestamp() {
+      this.db.set("lastTimestamp", new Date().toISOString()
+        .slice(0, 10));
+    },
+    _getLastTimestamp() {
+      return this.db.get("lastTimestamp");
+    },
+  },
+  hooks: {
+    async deploy() {
+      this._setLastTimestamp();
+
+      const response = await this.getResources();
+      const resources = response[this.getResourcesKey()];
+
+      resources.slice(10).reverse()
+        .forEach(this.emitEvent);
+    },
+  },
+  async run() {
+    const lastTimestamp = this._getLastTimestamp();
+    this._setLastTimestamp();
+
+    let page = 0;
+
+    while (true) {
+      const response = await this.getResources({
+        params: {
+          page,
+          after: lastTimestamp,
```

Comment:
I'm just wondering if this actually works because let's say a user made 1 sale and the source got trigger once so it gets one item in the list that is emitted, so the latest timestamp stored is for today, then another user makes another sale in the same day so that means it won't get any sale items for today because it will only get items after today date?
Or is it that after filter is inclusive?

Reply:
Yes, it's inclusive!